easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -134,6 +134,39 @@ def cache_aware_func(toolchain):
     return cache_aware_func
 
 
+def det_subtoolchain_version(current_tc, subtoolchain_name, optional_toolchains, cands):
+    """
+    Returns unique version for subtoolchain, in tc dict.
+    If there is no unique version:
+    * use '' for dummy, if dummy is not skipped.
+    * return None for skipped subtoolchains, that is,
+      optional toolchains or dummy without add_dummy_to_minimal_toolchains.
+    * in all other cases, raises an exception.
+    """
+    current_tc_name, current_tc_version = current_tc['name'], current_tc['version']
+
+    uniq_subtc_versions = set([subtc['version'] for subtc in cands if subtc['name'] == subtoolchain_name])
+    # init with "skipped"
+    subtoolchain_version = None
+
+    # dummy toolchain: bottom of the hierarchy
+    if subtoolchain_name == DUMMY_TOOLCHAIN_NAME:
+        if build_option('add_dummy_to_minimal_toolchains'):
+            subtoolchain_version = ''
+    elif len(uniq_subtc_versions) == 1:
+        subtoolchain_version = list(uniq_subtc_versions)[0]
+    elif len(uniq_subtc_versions) == 0:
+        if subtoolchain_name not in optional_toolchains:
+            # raise error if the subtoolchain considered now is not optional
+            raise EasyBuildError("No version found for subtoolchain %s in dependencies of %s",
```

Comment:
@bartoldeman I'm hitting this error with our `intel-para` toolchain but I don't understand why since `Iimkl` is an optional toolchain
```
== 2018-09-21 14:42:03,912 build_log.py:158 ERROR EasyBuild crashed with an error (at ?:124 in __init__): No version found for subtoolchain <class 'easybuild.toolchains.iimkl.Iimkl'> in dependencies of intel-para (at easybuild/framework/easyconfig/easyconfig.py:162 in det_subtoolchain_version)
```

Reply:
Never mind PR on the way